patches/1_70_0/0001-Beast-Fix-moved-from-executor-in-idle-ping-timeout.patch

```diff
@@ -0,0 +1,132 @@
+
+close #1599
+---
+ CHANGELOG.md                  |  6 +++++
+ test/beast/websocket/ping.cpp | 43 +++++++++++++++++++++++++++++++++++
+ 2 files changed, 49 insertions(+)
+
+diff --git a/CHANGELOG.md b/CHANGELOG.md
+index 43e39095..226997a7 100644
+--- a/CHANGELOG.md
++++ b/CHANGELOG.md
+@@ -1,3 +1,9 @@
++Version 248-hf1:
++
++* Add idle ping suspend test
++
++--------------------------------------------------------------------------------
++
+ Version 248:
+ 
+ * Don't use a moved-from handler
+diff --git a/test/beast/websocket/ping.cpp b/test/beast/websocket/ping.cpp
+index 7f63934f..1810927b 100644
+--- a/test/beast/websocket/ping.cpp
++++ b/test/beast/websocket/ping.cpp
+@@ -10,8 +10,11 @@
+ // Test that header file is self-contained.
+ #include <boost/beast/websocket/stream.hpp>
+ 
++#include <boost/beast/_experimental/test/tcp.hpp>
++
+ #include "test.hpp"
+ 
++#include <boost/asio/ip/tcp.hpp>
+ #include <boost/asio/io_context.hpp>
+ #include <boost/asio/strand.hpp>
+ 
+@@ -366,6 +369,46 @@ public:
+             BEAST_EXPECT(count == 3);
+         });
+ 
++        // suspend idle ping
++        {
++            using socket_type =
++                net::basic_stream_socket<
++                    net::ip::tcp,
++                    net::executor>;
++            net::io_context ioc;
++            stream<socket_type> ws1(ioc);
++            stream<socket_type> ws2(ioc);
++            ws1.set_option(stream_base::timeout{
++                stream_base::none(),
++                std::chrono::seconds(0),
++                true});
++            test::connect(
++                ws1.next_layer(),
++                ws2.next_layer());
++            ws1.async_handshake("localhost", "/",
++                [](error_code){});
++            ws2.async_accept([](error_code){});
++            ioc.run();
++            ioc.restart();
++            flat_buffer b1;
++            auto mb = b1.prepare(65536);
++            std::memset(mb.data(), 0, mb.size());
++            b1.commit(65536);
++            ws1.async_write(b1.data(),
++                [&](error_code, std::size_t){});
++            BEAST_EXPECT(
++                ws1.impl_->wr_block.is_locked());
++            ws1.async_read_some(net::mutable_buffer{},
++                [&](error_code, std::size_t){});
++            ioc.run();
++            ioc.restart();
++            flat_buffer b2;
++            ws2.async_read(b2,
++                [&](error_code, std::size_t){});
++            ioc.run();
++        }
++        //);
++
+         {
+             echo_server es{log, kind::async};
+             net::io_context ioc;
+-- 
+2.20.1.windows.1
+
+
+From 9e9c0f32c04f07934fc69986cfaac934b1705108 Mon Sep 17 00:00:00 2001
+From: Vinnie Falco <vinnie.falco@gmail.com>
+Date: Sat, 4 May 2019 15:31:11 -0700
+Subject: [PATCH 2/2] Fix moved-from executor in idle ping timeout
+
+fix #1599
+---
+ CHANGELOG.md                                | 1 +
+ include/boost/beast/websocket/impl/ping.hpp | 3 ++-
```

Comment:
The patch should be relative to the Boost release distribution: so the paths would start with   `boost/beast/` for headers, and `libs/beast/test/` for test sources.

Reply:
For examples, see:
* https://www.boost.org/patches/1_65_0/0004-thread-130-Bug-in-boost-condition_variable-on-Window.patch 
* https://www.boost.org/patches/1_65_0/0002-fiber-resume_with-requires-tor-return-a-continuation.patch